pkg/util/labels.go

```diff
@@ -60,27 +60,50 @@ func addDeploymentConfigNestedLabels(obj *deployapi.DeploymentConfig, labels lab
 	return nil
 }
 
+// addUnstructuredNestedLabels adds new label(s) to a nested labels of a single unstructured object
+func addUnstructuredNestedLabels(obj *runtime.Unstructured, labels labels.Set) error {
+	if template, ok := obj.Object["template"]; ok {
```

Comment:
Wouldn't it be more preferable if you checked for all the `!ok` cases instead and return an error in such cases? As is, if the passed unstructured object isn't a template or it doesn't have a nested controllerTemplate, podTemplate and so on, `addUnstructuredNestedLabels` won't add any labels to any template but still will not return any error as I believe it should. One more plus is that this change will de-arrowize the code ie. make it more readable.